lib/clients/http/tracing.axios-interceptor.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  AxiosFulfilledInterceptor,
+  AxiosInterceptor,
+  AxiosRejectedInterceptor,
+  AxiosResponseCustomConfig,
+} from "@narando/nest-axios-interceptor";
+import { HttpService, Injectable } from "@nestjs/common";
+import { Subsegment } from "aws-xray-sdk";
+import { getCauseTypeFromHttpStatus } from "aws-xray-sdk-core/lib/utils";
+import { AxiosRequestConfig } from "axios";
+import { ClientRequest, IncomingMessage } from "http";
+import { TracingService } from "../../core";
+import { TracingNotInitializedException } from "../../exceptions";
+import { HEADER_TRACE_CONTEXT } from "./http-tracing.constants";
+
+export const TRACING_CONFIG_KEY = Symbol("kTracingAxiosInterceptor");
+
+export interface TracingConfig extends AxiosRequestConfig {
+  [TRACING_CONFIG_KEY]: {
+    subSegment: Subsegment;
+  };
+}
+
+@Injectable()
+export class TracingAxiosInterceptor extends AxiosInterceptor<TracingConfig> {
+  constructor(
+    httpService: HttpService,
+    private readonly tracingService: TracingService
+  ) {
+    super(httpService);
+  }
+
+  requestFulfilled(): AxiosFulfilledInterceptor<TracingConfig> {
+    super.requestFulfilled;
```

Comment:
should probably be `super.requestFulfilled();`?

Reply:
This line was accidentally included and does not serve any purpose. I removed it.